spyder/utils/syntaxhighlighters.py

```diff
@@ -910,6 +910,130 @@ class HtmlSH(BaseWebSH):
+def make_md_patterns():
+    h1 = '^#[^#]+'
+    h2 = '^##[^#]+'
+    h3 = '^###[^#]+'
+    h4 = '^####[^#]+'
+    h5 = '^#####[^#]+'
+    h6 = '^######[^#]+'
+
+    titles = any('title', [h1, h2, h3, h4, h5, h6])
+
+    html_tags = any("builtin", [r"<", r"[\?/]?>", r"(?<=<).*?(?=[ >])"])
+    html_symbols = '&[^; ].+;'
+    html_comment = '<!--.+-->'
+
+    strikethrough = any('strikethrough', [r'(~~)(.*?)~~'])
+    strong = any('strong', [r'(\*\*)(.*?)\*\*'])
+
+    italic = r'(__)(.*?)__'
+    emphasis = r'(//)(.*?)//'
+    italic = any('italic', [italic, emphasis])
+
+    # links - (links) after [] or links after []:
+    link_html = (r'(?<=(\]\())[^\(\)]*(?=\))|'
+                 '(<https?://[^>]+>)|'
+                 '(<[^ >]+@[^ >]+>)')
+    # link/image references - [] or ![]
+    link = r'!?\[[^\[\]]*\]'
+    links = any('link', [link_html, link])
+
+    # blockquotes and lists -  > or - or * or 0.
+    blockquotes = (r'(^>+.*)'
+                   r'|(^(?:    |\t)*[0-9]+\. )'
+                   r'|(^(?:    |\t)*- )'
+                   r'|(^(?:    |\t)*\* )')
+    # code
+    code = any('code', ['^`{3,}.*$'])
+    inline_code = any('inline_code', ['`.*`'])
+
+    # math - $$
+    math = any('number', [r'^(?:\${2}).*$', html_symbols])
+
+    comment = any('comment', [blockquotes, html_comment])
+
+    return '|'.join([titles, comment, html_tags, math, links, italic, strong,
+                     strikethrough, code, inline_code])
+
+
+class MarkdownSH(BaseSH):
+    """Markdown Syntax Highlighter"""
+    # Syntax highlighting rules:
+    PROG = re.compile(make_md_patterns(), re.S)
+    NORMAL = 0
+    CODE = 1
+
+    def highlightBlock(self, text):
+        text = to_text_string(text)
+        previous_state = self.previousBlockState()
+
+        if previous_state == self.CODE:
+            self.setFormat(0, len(text), self.formats["code"])
+        else:
+            previous_state = self.NORMAL
+            self.setFormat(0, len(text), self.formats["normal"])
+
+        self.setCurrentBlockState(previous_state)
+
+        match = self.PROG.search(text)
+        while match:
+            for key, value in list(match.groupdict().items()):
+                start, end = match.span(key)
+
+                if value:
+                    previous_state = self.previousBlockState()
+
+                    if previous_state == self.CODE:
+                        if key == "code":
+                            # Change to normal
+                            self.setFormat(0, len(text),
+                                           self.formats["normal"])
+                            self.setCurrentBlockState(self.NORMAL)
+                        else:
+                            continue
+                    else:
+                        if key == "code":
+                            # Change to code
+                            self.setFormat(0, len(text), self.formats["code"])
+                            self.setCurrentBlockState(self.CODE)

```

Comment:
@ccordoba12 its a nice idea indeed, but I think we should postpone that to 4.0 as it would probably require a better re-estructuring of syntax highlighters.

I think for what we need this is already a good step